arangod/Aql/UnsortedGatherExecutor.cpp

```diff
@@ -79,102 +79,45 @@ auto UnsortedGatherExecutor::produceRows(typename Fetcher::DataRange& input,
     return {ExecutorState::DONE, Stats{}, AqlCallSet{}};
   } else {
     auto callSet = AqlCallSet{};
-    // TODO shouldn't we use `output.getClientCall()` instead of `AqlCall{}` here?
-    callSet.calls.emplace_back(AqlCallSet::DepCallPair{currentDependency(), AqlCall{}});
+    callSet.calls.emplace_back(
+        AqlCallSet::DepCallPair{currentDependency(), output.getClientCall()});
     return {input.upstreamState(currentDependency()), Stats{}, callSet};
   }
 }
 
 auto UnsortedGatherExecutor::skipRowsRange(typename Fetcher::DataRange& input, AqlCall& call)
     -> std::tuple<ExecutorState, Stats, size_t, AqlCallSet> {
-  // TODO call input.skip()!
-  while (call.needSkipMore() && !done()) {
-    if (input.hasDataRow(currentDependency())) {
-      auto [state, inputRow] = input.nextDataRow(currentDependency());
-      call.didSkip(1);
-
-      if (state == ExecutorState::DONE) {
-        advanceDependency();
-      }
-    } else {
-      if (input.upstreamState(currentDependency()) == ExecutorState::DONE) {
-        advanceDependency();
-      } else {
-        // We need to fetch more first
-        break;
-      }
-    }
-  }
-
-  while (!done() && input.upstreamState(currentDependency()) == ExecutorState::DONE) {
-    advanceDependency();
-  }
+  auto skipped = size_t{0};
 
-  if (done()) {
-    return {ExecutorState::DONE, Stats{}, call.getSkipCount(), AqlCallSet{}};
-  } else {
-    // TODO Should we not generate a call that only skips, with a soft limit of 0,
-    //      based off `call` rather than just asking for everything?
-    //      In which case of course it is important to look into the skip count
-    //      of the input ranges above!
-    auto callSet = AqlCallSet{};
-    callSet.calls.emplace_back(AqlCallSet::DepCallPair{currentDependency(), AqlCall{}});
-    return {input.upstreamState(currentDependency()), Stats{}, call.getSkipCount(), callSet};
-  }
-}
+  while (call.needSkipMore()) {
```

Comment:
If I'm not mistaken, this `while` loop always ends in a `return`, and after it sits a `TRI_ASSERT(false)`. So maybe
remove the loop and
```suggestion
  TRI_ASSERT(call.needSkipMore());
```
here instead?